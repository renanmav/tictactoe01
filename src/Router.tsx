import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';

import Menu from './screens/Menu';
import SelectDifficulty from './screens/SelectDifficulty';

export type MainStackParamList = {
  Menu: undefined;
  SelectDifficulty: undefined;
};

const MainStack = createStackNavigator<MainStackParamList>();

const Router: React.FC = () => {
  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="Menu"
        headerMode="none"
        screenOptions={{
          animationEnabled: false,
        }}>
        <MainStack.Screen name="Menu" component={Menu} />
        <MainStack.Screen name="SelectDifficulty" component={SelectDifficulty} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
};

export default Router;
