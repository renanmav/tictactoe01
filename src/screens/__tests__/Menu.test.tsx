import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react-native';

import Router from '../../Router';

test('examples of some things', async () => {
  const { getByText } = render(<Router />);

  const buttonJogar = getByText('Jogar');
  fireEvent.press(buttonJogar);

  await wait(() => expect(getByText('SelectDifficulty')).toMatchSnapshot());
});
