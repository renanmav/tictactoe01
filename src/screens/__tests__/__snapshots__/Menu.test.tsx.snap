// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<Text>
  SelectDifficulty
</Text>
`;
